src/borg/archiver/check_cmd.py

```diff
@@ -71,78 +71,114 @@ def build_parser_check(self, subparsers, common_parser, mid_common_parser):
-        until the complete repository has been checked. Example: Assuming a complete check took 7
-        hours, then running a daily check with --max-duration=3600 (1 hour) resulted in one
-        completed check per week.
-
-        Attention: A partial --repository-only check can only do way less checking than a full
-        --repository-only check: only the non-cryptographic checksum checks on segment file
-        entries are done, while a full --repository-only check would also do a repo index check.
-        A partial check cannot be combined with the ``--repair`` option. Partial checks
-        may therefore be useful only with very large repositories where a full check would take
-        too long.
-        Doing a full repository check aborts a partial check; the next partial check will restart
-        from the beginning.
-
-        The ``--verify-data`` option will perform a full integrity verification (as opposed to
-        checking the CRC32 of the segment) of data, which means reading the data from the
-        repository, decrypting and decompressing it. This is a cryptographic verification,
-        which will detect (accidental) corruption. For encrypted repositories it is
-        tamper-resistant as well, unless the attacker has access to the keys. It is also very
-        slow.
+        reliability of the hardware running Borg with stress testing software. This
+        especially includes storage and memory testers. Unreliable hardware might lead
+        to additional data loss.
+
+        It is highly recommended to create a backup of your repository before running
+        in repair mode (i.e. running it with ``--repair``).
+
+        Repair mode will attempt to fix any corruptions found. Fixing corruptions does
+        not mean recovering lost data: Borg can not magically restore data lost due to
+        e.g. a hardware failure. Repairing a repository means sacrificing some data
+        for the sake of the repository as a whole and the remaining data. Hence it is,
+        by definition, a lossy task.
+
+        In practice, repair mode hooks into both the repository and archive checks:
+
+        1. When checking the repository's consistency, repair mode will try to recover
+           as many objects from segments with integrity errors as possible, and ensure
+           that the index is consistent with the data stored in the segments.
+
+        2. When checking the consistency and correctness of archives, repair mode might
+           remove whole archives from the manifest if their archive metadata chunk is
+           corrupt or lost. On a chunk level (i.e. the contents of files), repair mode
+           will replace corrupt or lost chunks with a same-size replacement chunk of
+           zeroes. If a previously zeroed chunk reappears, repair mode will restore
+           this lost chunk using the new chunk. Lastly, repair mode will also delete
```

Comment:
Question: Can Borg only zero-out and restore *whole* chunks, or does Borg work with parts of chunks here?

If the first is true (my assumption), I'd like to avoid the word "healing" here. The reason is that we don't talk about files at this point (we talk about files later, that's also the reason for the redundancy, see below), but about chunks - and if my assumption was right, a chunk isn't healed, but replaced.

Thus I didn't make a change here. If I was wrong about my assumption (then it definitely needs paraphrasing), please tell me. If I wasn't wrong but you want it changed nevertheless (the way you suggested, or otherwise), please also just tell me, I'll change it then.

Reply:
I interpreted "places" as synonym for "places of files" - which can indeed be used as synonym for chunks. Sorry about that. However, I'd still like to avoid using "healing" in the context of chunks. Anyway, it's your decision, just tell me to change it to "damaged places" instead and I'll do so.

> If a file is damaged at multiple places (multiple chunks), it of course needs to heal all of them for a complete recovery.

Good point, I've tried to incorporate it below.